src/util/Views.h

```diff
@@ -129,18 +104,69 @@ CPP_template(typename UnderlyingRange, bool supportConst = true)(
     return ql::ranges::size(underlyingRange_);
   }
 
-  CPP_auto_member constexpr auto CPP_fun(data)()(
-      requires ql::ranges::contiguous_range<UnderlyingRange>) {
+  CPP_auto_member constexpr auto
+      CPP_fun(data) ()(requires ql::ranges::contiguous_range<UnderlyingRange>) {
     return ql::ranges::data(underlyingRange_);
   }
 
-  CPP_auto_member constexpr auto CPP_fun(data)()(
+  CPP_auto_member constexpr auto CPP_fun(data) ()(
       const  //
       requires ql::ranges::contiguous_range<const UnderlyingRange>) {
     return ql::ranges::data(underlyingRange_);
   }
 };
 
+// Takes a view of blocks and yields the elements of the same view, but removes
+// consecutive duplicates inside the blocks and across block boundaries.
+template <typename SortedBlockView,
+          typename BlockType = ql::ranges::range_value_t<SortedBlockView>,
+          typename ValueType = ql::ranges::range_value_t<BlockType>>
+
+auto uniqueBlockView(SortedBlockView view) {
```

Comment:
You can explicitly specify the return type here (it is `InputRangeTypeErased<BlockType>`